examples/siro_sandbox/app_states/app_state_fetch.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import magnum as mn
+import numpy as np
+from app_states.app_state_abc import AppState
+from camera_helper import CameraHelper
+from controllers.gui_controller import GuiHumanoidController
+from gui_navigation_helper import GuiNavigationHelper
+from hablab_utils import get_agent_art_obj_transform, get_grasped_objects_idxs
+
+from habitat.gui.gui_input import GuiInput
+from habitat.gui.text_drawer import TextOnScreenAlignment
+
+
+class AppStateFetch(AppState):
+    def __init__(
+        self,
+        sandbox_service,
+        gui_agent_ctrl,
+    ):
+        self._sandbox_service = sandbox_service
+        self.gui_agent_ctrl = gui_agent_ctrl
+
+        self._can_grasp_place_threshold = (
+            self._sandbox_service.args.can_grasp_place_threshold
+        )
+
+        self.cam_transform = None
+
+        self._held_target_obj_idx = None
+
+        # will be set in on_environment_reset
+        self._target_obj_ids = None
+
+        self._camera_helper = CameraHelper(
+            self._sandbox_service.args, self._sandbox_service.gui_input
+        )
+        self._first_person_mode = self._sandbox_service.args.first_person_mode
```

Comment:
We may potentially use self._camera_helper._first_person_mode (may consider making this attribute public).

Reply:
In my mind, these are somewhat separate. The AppState's first-person mode dictates user controls, which would probably be true even if we swapped out the camera in the future and even if we didn't have a camera.